odoo/tools/misc.py

```diff
@@ -1821,6 +1821,64 @@ def verify_hash_signed(env, scope, payload):
     return None
 
 
+def limited_field_access_token(record, field_name, timestamp=None):
```

Comment:
To me, if a function is defined inside the tools, then it should be a function for everybody, not only the models. Here it is using `record._name`, `record.id` and `record.env` (for hmac, which is another function that IMO shouldn't be in the tools).

I would much more rather have a generic function inside the tools:
```suggestion
def limited_field_access_token(unique_name: Any, hmac_secret: bytes, timestamp=None):
    if not timestamp:
        ...
    
    scope = 'binary'
    message = repr((scope, unique_name)).encode()
    token = hmac_lib.new(hmac_secret, message, hashlib.sha256).hexdigest()
    return f"{token}o{timestamp}"
```

with a counter-part defined on some model, e.g. `ir.binary`, or BaseModel, or a mixin, *flashback a previous discussion about bloating BaseModel with more stuff*[^1]

```py
def _limited_field_access_token(self, record, field_name):
    return limited_field_access_token(
        (record._name, record.id, field_name),
        env['ir.config_parameter'].get_param('database.secret')
    )
```

[^1]: maybe we should have `odoo.orm.tools.BaseTools`, a class imported by `Model` and `TransientModel` where all those tools would be defined. To avoid bloating the `BaseModel` class/file itself, and to still let all those tools be defined somewhere sensitive usable everywhere using `self.<method>`